### DISCUSSION TITLE
Running into issues with base64 encoding my App Store Connect API key

### DISCUSSION THREAD

**Discussion by ugiacoman** [fastlane/fastlane#17930] [Q&A] [answered]
I'm trying to setup the `app_store_connect_api_key` action with the code below. All the values are set through environment variables. API_KEY_CONTENT always fails though. I've tried the `/n` technique to no avail so I'd like to base64 encode. When I do, I get a null bytes error which leads me to believe I've encoded it incorrectly. After you download the API key .p8 file from App Store Connect, what are the steps to encoding that and setting it as an environment variable? Thanks!

```ruby
api_key = app_store_connect_api_key(
  key_id: API_KEY_ID,
  issuer_id: ISSUER_KEY_ID,
  key_content: API_KEY_CONTENT,
  in_house: false
)
```

**Comment by ugiacoman** [comment]
@joshdholtz I know you've recently added this support to fastlane. How did you go about doing this?

**Comment by rogerluan** [comment, marked as answer]
Hey @ugiacoman 👋 

>I've tried the /n technique to no avail

It's `\n`, not `/n`, maybe that was your issue? 

Also, how did you try base64-encoding it? AFAIK it should be simply `cat myfile.p8 | base64`

**Reply by jineshqa** [reply]
Call me naive but this `cat myfile.p8 | base64` should be inserted somewhere in the documentation [here](http://docs.fastlane.tools/actions/app_store_connect_api_key/#app_store_connect_api_key). Took me a couple of hours trying `\n`, `\\n`, collapsing key into a single line, etc.

**Comment by ugiacoman** [comment]
Ah, typo 🤦  I did use `\n` when testing, however simply encoding like you said worked. Thank you!
